src/Vocup/MainForm.cs

```diff
@@ -570,35 +567,34 @@ private bool EnsureSaved()
 
     private bool SaveFile(bool saveAsNewFile)
     {
-        //Datei-Speichern-unter Dialogfeld öffnen
+        BookFileFormat format = BookFileFormat.Vhf2;
+
         if (string.IsNullOrWhiteSpace(CurrentBook.FilePath) ||
             string.IsNullOrWhiteSpace(CurrentBook.VhrCode) ||
             saveAsNewFile)
         {
-            using (SaveFileDialog save = new SaveFileDialog
+            using SaveFileDialog save = new()
             {
                 Title = Words.SaveVocabularyBook,
                 FileName = CurrentBook.MotherTongue + " - " + CurrentBook.ForeignLang,
                 InitialDirectory = Program.Settings.VhfPath,
-                Filter = Words.VocupVocabularyBookFile + " (*.vhf)|*.vhf"
-            })
+                Filter = $"{Words.FileFormatVhf2} (*.vhf)|*.vhf|{Words.FileFormatVhf1} (*.vhf)|*.vhf"
```

Comment:
Why exactly did you choose to keep the same file extension for both the old and the new format? This might be confusing as from the filename you can't tell which version it is. Like MS Word uses .doc for the old file type (legacy) and .docx for the new filetype. The application can read and write to both versions and the file extension gives away what type of file it is.

Reply:
In my opinion it is more convenient for a user to see an error message that they must update Vocup when opening a new file than if that file would not be recognized as it has a new extension. I agree that it is not easy for users to see which version a .vhf file has, but I do not expect saving files in an old format will be a frequent use case. Also in the future there will be new breaking changes to .vhf file format and I do not want to use a new extension each time.

Reply:
I can follow this to a certain degree. But still, the internal file structure is totally different from the v1 version, what qualifies using a new extension to me. It's not necessarily true that you'll need a new extension for each breaking change as the file structure can be the same (zipped JSON format instead of the old binary) but the internal JSON should be structured in a way that new features are not breaking the internal file structure. As long as the file remains a zip archive with JSON inside that can be parsed by the logic, even adding new optional fields inside the JSON should'nt break the old logic that isn't aware of the new data. The same is somewhat happening when saving to the v1 fileformat and the new practicemode isn't recognized so you use a fallback. Also, when you insist on re-using the current extension, you can consider to not be able to save to the old format as this isn't really needed then.

Reply:
New file formats will likely keep the ZIP container but add breaking changes to the book JSON file for example when supporting any number of synonyms and therefore turning a `string` into a `List<Synonym>`.

I agree that the feature to save old file versions is not really required. It is intended for power users who have different versions of Vocup e.g. on Linux with Wine. For those users it is no problem to open a .vhf file in a text editor or archive tool to check its version. In my opinion saving a file in Vhf1 format is well hidden enough for most users not to play around with it.

Reply:
Still a new file extension (.vhf2) would fine as you still have a header file in JSON which should be parsable at all times. Then you can determine if the file format and thus the Book JSON should be parsable or to show a message that a newer version is required. In the end it's your call, be to me this seems more logical. The power users can easily distinguish the old versions from the new versions. The regular users probably won't even notice it.